mapiproxy/servers/default/emsmdb/emsmdbp_object.c

```diff
@@ -1234,6 +1234,68 @@ enum MAPISTATUS emsmdbp_folder_get_folder_count(struct emsmdbp_context *emsmdbp_
 	return retval;
 }
 
+static inline enum MAPISTATUS emsmdbp_object_move_folder_to_mapistore_root(struct emsmdbp_context *emsmdbp_ctx, struct emsmdbp_object *folder, struct emsmdbp_object *parent_folder, const char *new_name)
+{
+	uint64_t		parent_fid, fid, test_folder_id;
+	enum MAPISTATUS		retval;
+	char			*mapistore_uri;
+	TALLOC_CTX		*mem_ctx;
+
+
+	/* Sanity checks */
+	MAPI_RETVAL_IF(!emsmdbp_ctx, MAPI_E_INVALID_PARAMETER, NULL);
+	MAPI_RETVAL_IF(!folder, MAPI_E_INVALID_PARAMETER, NULL);
+	MAPI_RETVAL_IF(!parent_folder, MAPI_E_INVALID_PARAMETER, NULL);
+	MAPI_RETVAL_IF(!new_name, MAPI_E_INVALID_PARAMETER, NULL);
+
+	mem_ctx = talloc_zero(NULL, TALLOC_CTX);
```

Comment:
On Tue, Feb 10, 2015 at 09:40:34AM -0800, Jesús García Sáez wrote:
> > @@ -1234,6 +1234,68 @@ enum MAPISTATUS emsmdbp_folder_get_folder_count(struct emsmdbp_context *emsmdbp_
> >  	return retval;
> >  }
> >  
> > +static inline enum MAPISTATUS emsmdbp_object_move_folder_to_mapistore_root(struct emsmdbp_context *emsmdbp_ctx, struct emsmdbp_object *folder, struct emsmdbp_object *parent_folder, const char *new_name)
> > +{
> > +	uint64_t		parent_fid, fid, test_folder_id;
> > +	enum MAPISTATUS		retval;
> > +	char			*mapistore_uri;
> > +	TALLOC_CTX		*mem_ctx;
> > +
> > +
> > +	/* Sanity checks */
> > +	MAPI_RETVAL_IF(!emsmdbp_ctx, MAPI_E_INVALID_PARAMETER, NULL);
> > +	MAPI_RETVAL_IF(!folder, MAPI_E_INVALID_PARAMETER, NULL);
> > +	MAPI_RETVAL_IF(!parent_folder, MAPI_E_INVALID_PARAMETER, NULL);
> > +	MAPI_RETVAL_IF(!new_name, MAPI_E_INVALID_PARAMETER, NULL);
> > +
> > +	mem_ctx = talloc_zero(NULL, TALLOC_CTX);
> 
> To create a `TALLOC_CTX` it'd be better either `talloc_named` or `talloc_new`

Indeed.. This call is actually creating two talloc contexts.. :)

-- 
Jelmer Vernooij <jelmer@openchange.org> - https://jelmer.uk/